.circleci/config.yml

```diff
@@ -36,6 +39,16 @@ jobs:
           name: Lint with Black
           command: |
              poetry run black --diff --check .
+  typechecking:
+    executor: python/default
+    steps:
+      - checkout
+      - python/install-packages:
+          pkg-manager: poetry
+      - run:
+          name: Type check with mypy
+          command: |
+             poetry run mypy --ignore-missing-imports .
```

Comment:
without `--ignore-missing-imports`, typechecking failed like so:
```
% poetry run mypy .                                                               mypy
dags/tasks/symphony/request.py:3: error: Cannot find implementation or library stub for module named "airflow.providers.http.operators.http"
dags/tasks/folio/request.py:2: error: Cannot find implementation or library stub for module named "airflow.providers.http.operators.http"
dags/tasks/symphony/new.py:5: error: Cannot find implementation or library stub for module named "airflow.providers.http.operators.http"
dags/tasks/symphony/login.py:5: error: Cannot find implementation or library stub for module named "airflow.providers.http.operators.http"
dags/tasks/folio/login.py:5: error: Cannot find implementation or library stub for module named "airflow.providers.http.operators.http"
dags/tasks/amazon/s3.py:3: error: Cannot find implementation or library stub for module named "airflow.providers.amazon.aws.hooks.s3"
dags/tasks/amazon/s3.py:3: note: See https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports
dags/tasks/amazon/s3.py:4: error: Skipping analyzing "pymarc": found module but no type hints or library stubs
dags/cornell.py:5: error: Cannot find implementation or library stub for module named "folio"
dags/cornell.py:6: error: Cannot find implementation or library stub for module named "tasks.amazon.sqs"
dags/cornell.py:7: error: Cannot find implementation or library stub for module named "tasks.sinopia.sinopia"
dags/cornell.py:8: error: Cannot find implementation or library stub for module named "folio_request"
dags/cornell.py:9: error: Cannot find implementation or library stub for module named "folio_login"
dags/tasks/amazon/sqs.py:3: error: Cannot find implementation or library stub for module named "airflow.providers.amazon.aws.sensors.sqs"
dags/stanford.py:3: error: Cannot find implementation or library stub for module named "tasks.amazon.s3"
dags/stanford.py:4: error: Cannot find implementation or library stub for module named "tasks.amazon.sqs"
dags/stanford.py:5: error: Cannot find implementation or library stub for module named "tasks.sinopia.sinopia"
dags/stanford.py:6: error: Cannot find implementation or library stub for module named "tasks.sinopia.rdf2marc"
Found 17 errors in 9 files (checked 27 source files)
```

it'd be nice to get typechecking working on imports, but i figured a start was better than nothing, and didn't want to get distracted for too long by this isn't for a prioritized ticket.